src/commands.cc

```diff
@@ -412,30 +412,47 @@ static void ensure_all_buffers_are_saved()
     throw runtime_error(message);
 }
 
+template<bool force>
+void kill(const ParametersParser& parser, Context& context, const ShellContext&)
+{
+    auto& client_manager = ClientManager::instance();
+
+    if (not force and client_manager.count() == 1)
+        ensure_all_buffers_are_saved();
+
+    const int status = parser.positional_count() > 0 ? str_to_int(parser[0]) : 0;
+    while (not client_manager.empty())
+    {
+        auto client = client_manager.begin()->get();
+        client_manager.remove_client(*client, true, status);
```

Comment:
`**client_manager.begin()` would do, although I am tempted to add a method to ClientManager to clear all clients.

Reply:
I thought that using so many levels of indirection at once was a bit ugly and made the code less easy to understand, do you really want to get rid of that temporary variable?

Reply:
Yeah, I'd rather have the double indirection.